proteus/NumericalFlux.py

```diff
@@ -807,7 +807,11 @@ def setDirichletValues(self,ebqe):
             for (ebNE,k),g,x in zip(list(self.DOFBoundaryConditionsDictList[ci].keys()),
                                     list(self.DOFBoundaryConditionsDictList[ci].values()),
                                     list(self.DOFBoundaryPointDictList[ci].values())):
-                self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t)
+                try:
+                    self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t, numpy.array(self.vt.ebqe['n'][ebNE,k]))
+                except:
+                    self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t)
+                                        
```

Comment:
@cekees I have moved the try command here. It looks like now it works for all other models (VOF etc.) but still fails for Kappa model (while it shouldn't). The normals are not defined when Kappa calls this for the first time, so I get a NoneType error, as I have now set this value as default. Previously I was getting an infinite loop, as normals were set to 0. See trace below

Reply:
Traceback (most recent call last):
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalFlux.py", line 811, in setDirichletValues
    self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t, numpy.array(self.vt.ebqe['n'][ebNE,k]))
KeyError: 'n'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/HR/asd/PROTEUS/proteus//centos/bin/parun", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/HR/asd/PROTEUS/proteus/scripts/parun", line 601, in <module>
    runName + '_init_prof')
  File "/home/HR/asd/PROTEUS/proteus/proteus/Profiling.py", line 198, in __call__
    return func(*func_args, **func_kwargs)
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalSolution.py", line 614, in __init__
    self.allocateModels()
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalSolution.py", line 691, in allocateModels
    OneLevelTransportType=p.LevelModelType)
  File "/home/HR/asd/PROTEUS/proteus/proteus/Transport.py", line 6147, in __init__
    PhiSpaceTypeDict=phiSpaces)
  File "/home/HR/asd/PROTEUS/proteus/proteus/Transport.py", line 6374, in initialize
    movingDomain=movingDomain)
  File "/home/HR/asd/PROTEUS/proteus/proteus/mprans/Kappa.py", line 891, in __init__
    self.numericalFlux.setDirichletValues(self.ebqe)
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalFlux.py", line 813, in setDirichletValues
    self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t)
  File "proteus/mprans/BoundaryConditions.py", line 472, in mprans.BoundaryConditions.BC_RANS.setWallFunction.lambda3
  File "proteus/mprans/BoundaryConditions.py", line 1540, in mprans.BoundaryConditions.WallFunctions.get_k_dirichlet
  File "proteus/mprans/BoundaryConditions.py", line 1474, in mprans.BoundaryConditions.WallFunctions.tangentialVelocity
TypeError: unsupported operand type(s) for ** or pow(): 'NoneType' and 'int'

Reply:
I see that Kappa doesn't allocate/initialize normals. Were you able to grab it from the flow model as we discussed?

Reply:
Any more progress on this @adimako?